File: 프로그래머스/unrated/133502. 햄버거 만들기/햄버거 만들기.js
function solution(ingredient) {
    let count = 0;
    const stack = [];
    
    ingredient.forEach((item) => {
        stack.push(item);
        
        if (stack.length < 4) return;
        if (stack[stack.length - 1] !== 1) return;
        if (stack[stack.length - 2] !== 3) return;
        if (stack[stack.length - 3] !== 2) return;
        if (stack[stack.length - 4] !== 1) return;
        
        for (let i = 1; i <= 4; i++) {
            stack.pop();
        }
        count++;
    })
    return count;
}

File: 프로그래머스/1/133502. 햄버거 만들기/햄버거 만들기.py
def solution(ingredient):
    count = 0
    stack = []
    
    for i in ingredient:
        if len(stack) < 3:
            stack.append(i)
        else:
            if i == 1 and stack[-1] == 3 and stack[-2] == 2 and stack[-3] == 1:
                count += 1
                for _ in range(3):
                    stack.pop()
            else:
                stack.append(i)

    
    return count
                
        